bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/async/HeapSizeManager.java

```diff
@@ -63,61 +73,74 @@ public long getMaxHeapSize() {
     return maxHeapSize;
   }
 
-  public synchronized void waitUntilAllOperationsAreDone() throws InterruptedException {
+  public synchronized void flush() throws InterruptedException {
+    isFlushing.set(true);
     boolean performedWarning = false;
-    while(!pendingOperationsWithSize.isEmpty()) {
+    while (true) {
```

Comment:
I prefer the cleanup before the break.

   cleanupFinishedOperations();
    while (!pendingOperationsWithSize.isEmpty()) {
      if (!performedWarning && requiresWarning()) {
        warn();
        performedWarning = true;
      }
      wait(FINISH_WAIT_MILLIS);
      cleanupFinishedOperations();
    }